config/config.go

```diff
@@ -50,55 +79,79 @@ type Webserver struct {
 	SSLKey    env.String `toml:"ssl_key"`
 }
 
-// A Map represents a map in the Tegola Config file.
-type Map struct {
-	Name        env.String   `toml:"name"`
-	Attribution env.String   `toml:"attribution"`
-	Bounds      []env.Float  `toml:"bounds"`
-	Center      [3]env.Float `toml:"center"`
-	Layers      []MapLayer   `toml:"layers"`
-	TileBuffer  *env.Int     `toml:"tile_buffer"`
-}
+// ValidateAndRegisterParams ensures configured params don't conflict with existing
+// query tokens or have overlapping names
+func ValidateAndRegisterParams(mapName string, params []provider.QueryParameter) error {
+	if len(params) == 0 {
+		return nil
+	}
 
-// MapLayer represents a the config for a layer in a map
-type MapLayer struct {
-	// Name is optional. If it's not defined the name of the ProviderLayer will be used.
-	// Name can also be used to group multiple ProviderLayers under the same namespace.
-	Name          env.String `toml:"name"`
-	ProviderLayer env.String `toml:"provider_layer"`
-	MinZoom       *env.Uint  `toml:"min_zoom"`
-	MaxZoom       *env.Uint  `toml:"max_zoom"`
-	DefaultTags   env.Dict   `toml:"default_tags"`
-	// DontSimplify indicates whether feature simplification should be applied.
-	// We use a negative in the name so the default is to simplify
-	DontSimplify env.Bool `toml:"dont_simplify"`
-	// DontClip indicates whether feature clipping should be applied.
-	// We use a negative in the name so the default is to clipping
-	DontClip env.Bool `toml:"dont_clip"`
-	// DontClip indicates whether feature cleaning (e.g. make valid) should be applied.
-	// We use a negative in the name so the default is to clean
-	DontClean env.Bool `toml:"dont_clean"`
-}
+	usedNames := make(map[string]struct{})
```

Comment:
They do use `struct{}` e.g., [here](https://github.com/go-spatial/tegola/blob/master/internal/build/tags/tags.go#L124).

I strongly prefer `struct{}` as well, since this approach doesn't have invalid invariants  on type level i.e., `false` as a value in map. I just want a set of strings, any additional data is redundant.